gcloud-java-core/src/main/java/com/google/gcloud/ServiceFactory.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud;
+
+import java.io.Serializable;
+
+public interface ServiceFactory<
+        ServiceT extends Service, 
+        OptionsT extends ServiceOptions<?, OptionsT>>
+    extends Serializable {
```

Comment:
It is usually a mistake for interfaces to extend Serializable. The actual class that is serializable should do it. There is no benefit to user code to have the interface extend Serializable, and there are some implementation disadvantages. See Effective Java 2, Item 74.